test/python/circuit/test_circuit_operations.py

```diff
@@ -1263,11 +1263,11 @@ def test_pop_previous_instruction_removes_parameters(self):
         x, y = Parameter("x"), Parameter("y")
         test = QuantumCircuit(1, 1)
         test.rx(y, 0)
-        last_instructions = test.u(x, y, 0, 0)
+        last_instructions = list(test.u(x, y, 0, 0))
         self.assertEqual({x, y}, set(test.parameters))
 
         instruction = test._pop_previous_instruction_in_scope()
-        self.assertEqual(list(last_instructions), [instruction])
+        self.assertEqual(last_instructions, [instruction])
         self.assertEqual({y}, set(test.parameters))
```

Comment:
I think you can even undo this test change now to reduce the diff?

Reply:
This is actually a bit sneakier than it looks. The `_pop_previous_instruction_in_scope` is actually modifying `test._data`, and since `InstructionSet` now modifies `data` by index (via `_add_ref`), we have an issue.